token-metadata/program/tests/escrow.rs

```diff
@@ -0,0 +1,213 @@
+#![cfg(feature = "test-bpf")]
+pub mod utils;
+
+use solana_program::borsh::try_from_slice_unchecked;
+use solana_program_test::*;
+use solana_sdk::{signer::Signer, transaction::Transaction};
+use utils::*;
+
+mod escrow {
+    use mpl_token_metadata::{escrow::find_escrow_account, state::EscrowAuthority};
+    use solana_program::program_pack::Pack;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn smoke_test_success() {
+        let mut context = program_test().start_with_context().await;
+
+        // Create Escrow
+        print!("\n=====Create Escrow=====\n");
+        let parent_test_metadata = Metadata::new();
+        let parent_test_master_edition = MasterEditionV2::new(&parent_test_metadata);
+        parent_test_metadata
+            .create_v2(
+                &mut context,
+                "Test".to_string(),
+                "TST".to_string(),
+                "uri".to_string(),
+                None,
+                10,
+                true,
+                None,
+                None,
+                None,
+            )
+            .await
+            .unwrap();
+
+        parent_test_master_edition
+            .create_v3(&mut context, Some(0))
+            .await
+            .unwrap();
+
+        let escrow_address = find_escrow_account(
+            &parent_test_metadata.mint.pubkey(),
+            &EscrowAuthority::TokenOwner,
+        );
+        print!("\nEscrow Address: {:#?}\n", escrow_address);
+
+        let ix0 = mpl_token_metadata::escrow::create_escrow_account(
+            mpl_token_metadata::id(),
+            escrow_address.0,
+            parent_test_metadata.pubkey,
+            parent_test_metadata.mint.pubkey(),
+            parent_test_metadata.token.pubkey(),
+            parent_test_master_edition.pubkey,
+            context.payer.pubkey(),
+            None,
+        );
+        println!("{:?} {:?}", &context.payer, &parent_test_metadata.token);
+
+        let tx0 = Transaction::new_signed_with_payer(
+            &[ix0],
+            Some(&context.payer.pubkey()),
+            &[&context.payer],
+            context.last_blockhash,
+        );
+
+        context.banks_client.process_transaction(tx0).await.unwrap();
+
+        let _metadata = parent_test_metadata.get_data(&mut context).await;
+        let escrow_account = get_account(&mut context, &escrow_address.0).await;
+        let escrow: mpl_token_metadata::state::TokenOwnedEscrow =
+            try_from_slice_unchecked(&escrow_account.data).unwrap();
```

Comment:
This is fine for a test, but all `TokenMetadataAccount`s have the `safe_deserialize` function available which checks at least the key, to prevent accidentally trying to deserialize the wrong account type.